controllers/comments.controller.js

```diff
@@ -1,11 +1,43 @@
-const { selectCommentsByArticleId } = require("../models/comments-model");
+const { checkArticleIdExists } = require("../models/articles-model");
+const {
+  selectCommentsByArticleId,
+  insertNewComment,
+} = require("../models/comments-model");
+const { checkUserExists } = require("../models/users-model");
 
 exports.getArticleComments = (req, res, next) => {
   const { article_id } = req.params;
 
-  selectCommentsByArticleId(article_id)
+  checkArticleIdExists(article_id)
+    .then(() => {
+      return selectCommentsByArticleId(article_id);
+    })
     .then((comments) => {
       return res.status(200).send({ comments });
     })
     .catch(next);
 };
+
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const newComment = req.body;
+
+  if (!newComment.username || !newComment.comment) {
+    return res.status(400).send({ msg: "Something wrong with body" });
+  }
+
+  const promises = [
```

Comment:
These are both foreign keys so PSQL should give you some errors anyway that you can handle